### PR TITLE
Create CAN_PG_A-M.cup

### PR DESCRIPTION
**Pull request by ventus99**
What is the purpose of this change? What does it change?
--------------------------------------------------------
Adding waypoints and tasks for Proving Grounds https://soaringtasks.com/

New

What is the source of the data (for e.g. new frequencies)?
----------------------------------------------------------
Adding waypoints and tasks for Proving Grounds https://soaringtasks.com/

### CODE REVIEW

**Review by lordfolken — CHANGES_REQUESTED** [review]
Please be consistent in waypoint descriptions.

File: waypoints-special/CAN_PG_A-M.cup [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+"Bruderheim",Brudeeim,CA,5348.012N,11255.476W,629.7m,1,,,,
+"CAGC Point",CAGCPint,,5204.740N,11402.040W,919.6m,1,,,,
+"Casselman",CASSELMN,CA,4518.717N,07505.133W,64.0m,1,,,,"Turn Point, Rd. Int."
+"Chipman",Chipman,CA,5343.050N,11238.000W,669.0m,2,100,792.5m,"123.4","Turnpoint, (UNICOM o/t tfc).  Glider activity.  10/28."
+"Chokio",Chokio,CA,4934.391N,11339.798W,1055.8m,1,,,,"Turnpoint."
+"Conestogo Lake",16CNSTGL,CA,4340.518N,08043.002W,396.2m,1,,,,"Turn Point"
+"Connemara",Conneara,CA,5023.834N,11350.001W,1036.9m,1,,,,"Turnpoint."
+"Cornwall",CYCC,CA,4505.568N,07434.068W,53.3m,5,280,1069.8m,"122.775","ATF 5 nm 3200 asl 10/28 3510 x 100 asphalt "
+"Cowley a/p",Cowleyap,CA,4938.183N,11405.667W,1181.4m,2,110,1319.8m,"123.4","Turnpoint, (tfc).  Glider activity.  11/29, 03/21."
+"CSS4",STDO,CA,4537.950N,07249.133W,75.9m,2,150,590.1m,"122,475",
```
Comment (lordfolken) [inline waypoints-special/CAN_PG_A-M.cup:16]: Comma instead of .

**Review by lordfolken — CHANGES_REQUESTED** [review]
Some cleanup required, so it actually passes the parser. :-) :+1:

**Review by lordfolken — APPROVED** [review]
Good work. :-) :+1: 
Keep it up.

File: waypoints-special/CAN_PG_A-M.cup [hunk 2]
```diff
@@ -1,415 +1,414 @@
...
 "Two Hills",TwoHills,CA,5342.000N,11145.964W,596.0m,1,,,,
-"Vankleek Hill",VNKLKHLL,CA,4531.183N,07439.133W,115.8m,1,,,,"Turn Point, Rd. Int."
-"Vegreville",Vegrevil,CA,5330.867N,11201.650W,631.5m,5,130,1219.2m,"122.8","Turnpoint, (UNICOM o/t tfc).  Righthand circuits Rwy 31.  NOTE:  Gun club 1 nm NW.  Avoid or overfly no less than 1000 ft AGL.  Ultralight activity in vicinity.  13/31."
+"Vankleek Hill",VNKLKHLL,CA,4531.183N,07439.133W,115.8m,1,,,,
+"Vegreville",Vegrevil,CA,5330.867N,11201.650W,631.5m,5,130,1219.2m,"122.8",(UNICOM o/t tfc).  Righthand circuits Rwy 31.  NOTE:  Gun club 1 nm NW.  Avoid or overfly no less than 1000 ft AGL.  Ultralight activity in vicinity.  13/31."
 "Wimborne Hamlet",WINBOH,CA,5151.600N,11335.500W,946.0m,1,,,,
 "Woodstock",Woodsown,CA,4306.582N,08049.248W,295.7m,1,,,,
 -----Related Tasks-----
-"AVVC 1 - St Dominique","CSS4","St Dominique City","St Liboire","Ste Rosalie","St Dominique City","CSS4"
+"AVVC 1 - St Dominique","CSS4","St Dominique City","St Liboire","Ste Rosalie","St Dominique City","CSS4",,,,
```
Comment (lordfolken) [inline waypoints-special/CAN_PG_A-M.cup]: So here according to cup standard http://download.naviter.com/docs/CUP-file-format-description.pdf
The ,,,, are not necessary. Its just a task name and a list of turn points.

File: waypoints-special/CAN_PG_A-M.cup [hunk 3]
```diff
@@ -1,415 +1,414 @@
...
 Options,WpDis=False,NearDis=0.0km,NearAlt=0.0ft
-ObsZone=0,Style=2,R1=2000m,A1=180
 ObsZone=1,Style=1,R1=2000m,A1=180
 ObsZone=2,Style=1,R1=2000m,A1=180
-ObsZone=3,Style=3,R1=2000m,A1=180
+ObsZone=3,Style=3,R1=2000m,A1=180,Move=1
```
Comment (lordfolken) [inline waypoints-special/CAN_PG_A-M.cup:62]: I cannot find any reference to this Move=1 in the documentation. What does it do?

File: waypoints-special/CAN_PG_A-M.cup [hunk 4]
```diff
@@ -1,415 +1,414 @@
...
 "AVVC 1 - St Liboire","CSS4","St Liboire","Ste Rosalie","St Dominique City","St Liboire","CSS4"
 Options,WpDis=False,NearDis=0.0km,NearAlt=0.0ft
```
Comment (lordfolken) [inline waypoints-special/CAN_PG_A-M.cup:64]: Please use NearAlt=0.0m

File: waypoints-special/CAN_PG_A-M.cup [hunk 5]
```diff
@@ -1,415 +1,414 @@
...
 "AVVC 1inv - St Liboire","CSS4","St Liboire","St Dominique City","Ste Rosalie","St Liboire","CSS4"
 Options,WpDis=False,NearDis=0.0km,NearAlt=0.0ft
-ObsZone=0,Style=2,R1=2000m,A1=180
+ObsZone=0,Style=2,R1=2000m,A1=180,Move=1
 ObsZone=1,Style=1,R1=2000m,A1=180
 ObsZone=2,Style=1,R1=2000m,A1=180
-ObsZone=3,Style=3,R1=2000m,A1=180
-"AVVC 1inv - Ste Rosalie","CSS4","Ste Rosalie","St Liboire","St Dominique City","Ste Rosalie","CSS4"
+ObsZone=3,Style=3,R1=2000m,A1=180,Move=1
+"AVVC 1inv - Ste Rosalie","CSS4","Ste Rosalie","St Liboire","St Dominique City","Ste Rosalie","CSS4",,,,
```
Comment (lordfolken) [inline waypoints-special/CAN_PG_A-M.cup]: The format of this line is "Description","wp1","wp2" the commas at the end are not necessary and might confuse the parser.

File: waypoints-special/CAN_PG_A-M.cup [hunk 6]
```diff
@@ -1,415 +1,414 @@
...
 Options,WpDis=False,NearDis=0.0km,NearAlt=0.0ft
-ObsZone=0,Style=2,R1=2000m,A1=180
+ObsZone=0,Style=2,R1=2000m,A1=180,Move=1
 ObsZone=1,Style=1,R1=2000m,A1=180
 ObsZone=2,Style=1,R1=2000m,A1=180
 ObsZone=3,Style=1,R1=2000m,A1=180
-ObsZone=4,Style=3,R1=2000m,A1=180
-"AVVC 3","CSS4","CSS4","St Hyacinthe","Drummondville Airport","CSS4","CSS4"
+ObsZone=4,Style=3,R1=2000m,A1=180,Move=1
+"AVVC 3","CSS4","CSS4","St Hyacinthe","Drummondville Airport","CSS4","CSS4",,,,
```
Comment (lordfolken) [inline waypoints-special/CAN_PG_A-M.cup]: This is what the parser is falling over. The repeated waypoints. It cannot parse a course from the same location to the same location again.

File: waypoints-special/CAN_PG_A-M.cup [hunk 7]
```diff
@@ -1,415 +1,414 @@
...
-"MSC Racetrack rev","Hawkesbury","Hawkesbury","Trans Canada","Vankleek Hill","Blue Corners","Hawkesbury","Hawkesbury"
+"MSC Racetrack rev","Hawkesbury","Hawkesbury","Trans Canada","Vankleek Hill","Blue Corners","Hawkesbury","Hawkesbury",,,
 Options,WpDis=False,NearDis=0.0km,NearAlt=0.0ft
 ObsZone=0,Style=2,R1=2000m,A1=180,Move=1
 ObsZone=1,Style=1,R1=2000m,A1=180
 ObsZone=2,Style=1,R1=2000m,A1=180
 ObsZone=3,Style=1,R1=2000m,A1=180
 ObsZone=4,Style=3,R1=2000m,A1=180,Move=1
-"MSC Sprint","Hawkesbury","Hawkesbury","Pendleton","Alexandria","Hawkesbury","Hawkesbury"
+"MSC Sprint","Hawkesbury","Hawkesbury","Pendleton","Alexandria","Hawkesbury","Hawkesbury",,,,
```
Comment (lordfolken) [inline waypoints-special/CAN_PG_A-M.cup]: Another instance of repeated waypoints. Please correct all of them.

**Review by lordfolken — CHANGES_REQUESTED** [review]
The waypoints are fine now. There are some issues in the tasks:
* Move=1 i don't know what that option is doing. it does not exist in the cup file specification
* Repeated waypoints in task wp1,wp1,wp2,wp3,wp4,wp4
* Superflicous commas in task names:

File: waypoints-special/CAN_PG_A-M.cup (hunk 3 above)
Comment (ventus99) [reply waypoints-special/CAN_PG_A-M.cup:63]: I have an old version of SeeYou so maybe this is not supported anymore.  The Move=1 is to move the origin of the start.  If 0 it uses the center of the circle.  If 1 it uses the edge.

File: waypoints-special/CAN_PG_A-M.cup (hunk 4 above)
Comment (ventus99) [reply waypoints-special/CAN_PG_A-M.cup]: Done

File: waypoints-special/CAN_PG_A-M.cup (hunk 5 above)
Comment (ventus99) [reply waypoints-special/CAN_PG_A-M.cup]: Thank you.  I was switching to a comma delimited file and these were remnants.  I opened the file in SeeYou again and then saved.  That got rid of it.

File: waypoints-special/CAN_PG_A-M.cup (hunk 6 above)
Comment (ventus99) [reply waypoints-special/CAN_PG_A-M.cup]: At least in my version of SeeYou, the first CSS4 is the takeoff point and the second is the start.

File: waypoints-special/CAN_PG_A-M.cup (hunk 7 above)
Comment (ventus99) [reply waypoints-special/CAN_PG_A-M.cup]: At least in my version of SeeYou, the first Hawkesbury is the takeoff point and the second is the start.

File: waypoints-special/CAN_PG_A-M.cup (hunk 6 above)
Comment (ventus99) [reply waypoints-special/CAN_PG_A-M.cup]: And the last CSS4s are finish and landing point

File: waypoints-special/CAN_PG_A-M.cup (hunk 7 above)
Comment (lordfolken) [reply waypoints-special/CAN_PG_A-M.cup]: I observed this too now on my version of seeyou.

**Review by lordfolken — APPROVED** [review]

**Pull request merged**